vg/draw/canvas.go

```diff
@@ -576,29 +615,54 @@ func isect(p0, p1, clip, norm vg.Point) vg.Point {
 }
 
 // FillText fills lines of text in the draw area.
-// The text is offset by its width times xalign and
-// its height times yalign.  x and y give the bottom
-// left corner of the text befor e it is offset.
-func (c *Canvas) FillText(sty TextStyle, pt vg.Point, xalign, yalign float64, txt string) {
+// x and y specify the location where the text is to be drawn.
+func (c *Canvas) FillText(sty TextStyle, pt vg.Point, txt string) {
 	txt = strings.TrimRight(txt, "\n")
 	if len(txt) == 0 {
 		return
 	}
 
 	c.SetColor(sty.Color)
 
-	ht := sty.Height(txt)
-	pt.Y += ht*vg.Length(yalign) - sty.Font.Extents().Ascent
+	if sty.Rotation != 0 {
+		c.Push()
+		c.Rotate(sty.Rotation)
+	}
+
+	cos := vg.Length(math.Cos(sty.Rotation))
+	sin := vg.Length(math.Sin(sty.Rotation))
+	pt.X, pt.Y = pt.Y*sin+pt.X*cos, pt.Y*cos-pt.X*sin
+
 	nl := textNLines(txt)
+	ht := sty.Height(txt)
+	pt.Y += ht*vg.Length(sty.YAlign) - sty.Font.Extents().Ascent
 	for i, line := range strings.Split(txt, "\n") {
-		xoffs := vg.Length(xalign) * sty.Font.Width(line)
+		xoffs := vg.Length(sty.XAlign) * sty.Font.Width(line)
 		n := vg.Length(nl - i)
 		c.FillString(sty.Font, pt.Add(vg.Point{X: xoffs, Y: n * sty.Font.Size}), line)
 	}
+
+	if sty.Rotation != 0 {
+		c.Pop()
+	}
+}
+
+// rotatePoint applies rotation Θ (in radians) about the origin to point p.
+func rotatePoint(Θ float64, p vg.Point) vg.Point {
```

Comment:
Use the name of the letter rather than the greek. This is a bug waiting to happen.